commit a8fdfb0b39c2b31722c70bdf2272b949d5af4b7b
Author: Quinn Tran <qutran@marvell.com>
Date:   Wed Jun 8 04:58:45 2022 -0700

    scsi: qla2xxx: edif: Fix session thrash
    
    Current code prematurely sends out PRLI before authentication application
    has given the OK to do so. This causes PRLI failure and session teardown.
    
    Prevents PRLI from going out before authentication app gives the OK.
    
    Link: https://lore.kernel.org/r/20220608115849.16693-7-njavali@marvell.com
    Fixes: 91f6f5fbe87b ("scsi: qla2xxx: edif: Reduce connection thrash")
    Reviewed-by: Himanshu Madhani <himanshu.madhani@oracle.com>
    Signed-off-by: Quinn Tran <qutran@marvell.com>
    Signed-off-by: Nilesh Javali <njavali@marvell.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/qla2xxx/qla_edif.c b/drivers/scsi/qla2xxx/qla_edif.c
index 17dcc4a32118..51736a71f2b6 100644
--- a/drivers/scsi/qla2xxx/qla_edif.c
+++ b/drivers/scsi/qla2xxx/qla_edif.c
@@ -3517,7 +3517,7 @@ int qla_edif_process_els(scsi_qla_host_t *vha, struct bsg_job *bsg_job)
 	if (qla_bsg_check(vha, bsg_job, fcport))
 		return 0;
 
-	if (fcport->loop_id == FC_NO_LOOP_ID) {
+	if (EDIF_SESS_DELETE(fcport)) {
 		ql_dbg(ql_dbg_edif, vha, 0x910d,
 		    "%s ELS code %x, no loop id.\n", __func__,
 		    bsg_request->rqst_data.r_els.els_code);